spec/The-OpenFF-System.md

```diff
@@ -0,0 +1,185 @@
+
+### Extensibility to new potentials
+
+The base class for potential handlers specifies the minimum amount of information required to write a new handler. If a user wishes to write a custom handler, they can either subclass out of this base class or instantiate it directly with a custom set of data and parametrization rules. For example, a handler encoding the Buckingham potential could be created mostly by copying a Lennard-Jones non-bonded handler and modifying the `expression` attribute (assuming that the parametrization machinery used for the Lennard-Jones-based terms can safely drop in).  Handlers storing more novel terms, such as CMAPs, will require more custom code but will fit into the framework of the base class.
+
+### Partial re-parametrization
+
+Some handlers may encode special triggers for re-parametrization. For example, consider a `ProperTorsionHandler` that relies on bond order interpolation. This typing can depends on a fractional bond order assignment method. If that is modified, the parametrization (of only one handler) could be re-triggered.
+
+### Electrostatics meta-handler
+
+Potential handlers are meant to be highly modular with lifecycles that act independently of all others, i.e. the entire parametrization process, data structures, and utilities of the bonds handler exist independently of the angle handler.  Electrostatics cannot behave this way, as the final parametrization (a set of partial charges) is often the sum of smaller handlers (a combination of library charges, partial charge assignment methods, charge increments, etc.) These handlers, not only the partial charges, need to be accessible to optimization routines and user inspection.
+
+### Batch modification of parameters
+
+Some amount (all?) of parameters will be exposed to modification by the user.
+
+Say, for example, one wanted to scale all partial charges in a system by a constant factor.
+
+```python3
+factor = 0.8
+electrostatics = system.get_handler('Electrostatics')
+for potential in electrostatics.potentials:
+    potential.charge *= factor
+```
+
+Each handler will need to define its own rules for what modifications are allowed and, after modification, what status check(s) must pass.
+
+### User control over system combination
+
+A plug-in architecture will expose settings for the level of rigour executed in the internal consistency checks. The default will be strict, resulting in errors if there is any ambiguity in the physical description of each system (i.e. different non-bonded cutoff treatments). A small number (1-2) of other settings with more granularity will be exposed, i.e. one that is problematively permissive and another that implements some amount of "reasonable" discrepancies to be fudged together. This will all be constructed in a way that enables users to define their own sets of "knobs" for when system combination can and cannot be allowed, i.e. if aromaticity models need not match but cut-off treatments must.
```

Comment:
Maybe there can be compositions of functions?  It would be a pain to write out different van der Waals, PME electrostatic short range, etc. that each had to reimplement a tapered cutoff term to multiply by each term - seems like should only be written once.

Reply:
Also, is it possible to recognize when two functions are equivalent (At least 2 different ways of doing LJ, harmonic force constants can be defined with or without 1/2, RB and periodic sums of torsion can be equivalent).